src/languages/php.js

```diff
@@ -263,13 +262,65 @@ export default function(hljs) {
     "stdClass"
   ];
 
+  /** Dual-case keywords
+   *
+   * ["then","FILE"] =>
+   *     ["then", "THEN", "FILE", "file"]
+   *
+   * @param {string[]} items */
+  const dualCase = (items) => {
+    /** @type string[] */
+    const result = [];
+    items.forEach(item => {
+      result.push(item);
+      if (item.toLowerCase() === item) {
+        result.push(item.toUpperCase());
+      } else {
+        result.push(item.toLowerCase());
+      }
+    });
+    return result;
+  };
+
   const KEYWORDS = {
-    keyword: KWS,
-    literal: LITERALS,
-    built_in: BUILT_INS
+    keyword: KWS.concat([ "match|0" ]),
+    literal: dualCase(LITERALS),
+    built_in: BUILT_INS.concat([ "Error|0" ]),
+  };
+
+  const CONSTRUCTOR_CALL = {
+    variants: [
+      {
+        match: [
+          /new/,
+          /\s+/,
+          /\\?\w+/,
+          /\s*\(/,
+        ],
+        scope: {
+          1: "keyword",
+          3: "title.class",
+        },
+      }
+    ]
+  };
+
+  const FUNCTION_INVOKE = {
+    relevance: 0,
+    match: [
+      /(?:->|::|\s|\(|\\)/,
+      regex.concat("(?!fn\\b|function\\b|match\\b|Error\\b|", KWS.join("\\b|"), "|", BUILT_INS.join("\\b|"), "\\b)"),
```

Comment:
How many of the builtins and keywords are truly used like functions?  Is it a majority?  In most languages it's only a few... if there was a list of 5-10 in PHP I'd rather be explicit rather than just negative match the ENTIRE world of keywords (which does have a cost).

Reply:
Or if we just added an explicit rule for `[BUILT_IN](` (if many are used like functions) that would also remove the need for all the negative matching of built-ins here...

Reply:
Also if you are doing negative filters then  you need a few markup tests to test the negative case as well to show that your negative filter is working as intended (you can add comments to explain the purposes of the negative tests).

Reply:
> How many of the builtins and keywords are truly used like functions? Is it a majority? In most languages it's only a few... if there was a list of 5-10 in PHP I'd rather be explicit rather than just negative match the ENTIRE world of keywords (which does have a cost).




```
  const MIGHT_BE_BEFORE_OPENING_BRACKET = [
    "array",
    "case",
    "catch",
    "die",
    "echo",
    "exit",
    "include",
    "include_once",
    "print",
    "require",
    "require_once",
    "do",
    "else",
    "elseif",
    "empty",
    "eval",
    "finally",
    "for",
    "foreach",
    "fn",
    "function",
    "if",
    "int",
    "isset",
    "list",
    "match",
    "return",
    "switch",
    "throw",
    "trait",
    "try",
    "unset",
    "use",
    "while",
  ];
```
This was my first approach ^.
There will be also some from `BUILT_INS`.



> Also if you are doing negative filters then you need a few markup tests to test the negative case as well to show that your negative filter is working as intended (you can add comments to explain the purposes of the negative tests).

I reused old tests. New code didn't break them, but some comment would be in place. I will add some extra markup tests

Reply:
> Also if you are doing negative filters then you need a few markup tests to test the negative case as well to show that your negative filter is working as intended (you can add comments to explain the purposes of the negative tests).

 @joshgoebel Added extra tests and some comments.